File: seedWork/SeedWork.Domain/ValueObjects/FullName.cs
using Shared.Domain;

namespace SeedWork.Domainn.ValueObjects;

/// <summary>
/// ФИО.
/// </summary>
public class FullName : ValueObject
{
    /// <summary>
    /// Имя.
    /// </summary>
    public required string Name { get; init; }

    /// <summary>
    /// Фамилия.
    /// </summary>
    public required string Surname { get; init; }

    /// <summary>
    /// Отчество.
    /// </summary>
    public string? Patronymic { get; init; }

    protected override IEnumerable<object?> GetEqualityComponents()
    {
        yield return Name;
        yield return Surname;
        yield return Patronymic;
    }
}

File: ui/eas-web/src/sharedModels/FullName.ts

export class FullName {
    constructor(
        public readonly name: string,
        public readonly surname: string,
        public readonly patronymic?: string) { }

    public ToString(): string {
        const s = this.surname || '';
        const f = this.name || '';
        if (this.patronymic) {
            return `${s} ${f} ${this.patronymic}`;
        }
        return `${s} ${f}`;
    }
}